shader/graph/src/link.rs

```diff
@@ -122,7 +122,7 @@ impl ShaderControlFlowNode {
     // to the parent scope
     for write in writes {
       let im_write = ShaderGraphNode::Write {
-        old: None,
+        old: Some(node.handle()),
```

Comment:
ShaderGraphNode::Write' old means the value that has been overwritten. This write node is handling the mutation of cfg struct to the outer scope, and it is not written to any value(actually has, but we are hard to know at this point). But this write node should still depend on cfg node. To express this dependency, instead of using cfg node as `old`, try to do a direct node connecting here

Reply:
where is this direct node come form?

Reply:
I mean directly connect cfg node with this im_write node to create dependency